fastlane/Helpers/Match.swift

```diff
@@ -26,24 +26,72 @@ enum Match {
             match(
                 type: type.value,
                 readonly: .userDefined(!isForce),
-                appIdentifier: appIdentifier,
+                appIdentifier: [environment.bundleId],
                 username: .userDefined(Constant.userName),
                 teamId: .userDefined(Constant.teamId),
                 gitUrl: Constant.matchURL,
                 force: .userDefined(isForce)
             )
         }
+        updateCodeSigning(type: type, environment: environment)
+    }
+    
+    static func updateCodeSigning(type: MatchType, environment: Environment) {
+        // Update Code signing from automatic to manual
+        updateCodeSigningSettings(
+            path: Constant.projectPath,
+            useAutomaticSigning: .userDefined(false),
+            teamId: .userDefined(Constant.teamId),
+            targets: .userDefined([Constant.projectName]),
+            buildConfigurations: .userDefined(["\(type.buildConfiguration) \(environment.rawValue)"]),
+            codeSignIdentity: .userDefined(type.codeSignIdentity),
+            profileName: .userDefined("match \(type.method) \(environment.bundleId)")
+        )
     }
 }
 
 extension Match {
-
+    
+    enum Environment: String {
+        
+        case staging = "Staging"
+        case production = "Production"
+        
+        var bundleId: String {
+            switch self {
+            case .staging: return Constant.stagingBundleId
+            case .production: return Constant.productionBundleId
+            }
+        }
+    }
+    
     enum MatchType: String {
 
         case development
         case adHoc = "adhoc"
         case appStore = "appstore"
 
         var value: String { return rawValue }
+
+        var method: String {
+            switch self {
+            case .development: return "Development"
+            case .adHoc: return "AdHoc"
+            case .appStore: return "AppStore"
+            }
+        }
+        
+        var buildConfiguration: String {
+            switch self {
+            case .development: return "Debug"
```

Comment:
Please double check, our build configurations:
- Debug Staging
- Release Staging
- Debug Production
- Release Production

Reply:
Please check because Ed's code is to combine MatchType build configuration with Environment raw.

`buildConfigurations: .userDefined(["\(type.buildConfiguration) \(environment.rawValue)"])`